### PR TITLE
Deploy: ignore errors produced by "dd" when creating the remote shell script for push deployment

### PR DESCRIPTION
**Pull request by rwaldron**
For some reason that I cannot figure out, creating the shell script with `openStdinToFile(...)` was failing to "close". Digging in I discovered that this output was being produced: 

```
0+1 records in
0+1 records out
```

Despite that, the file was being created and written to as expected. 

Other places that openStdinToFile is used:

- t2 rename: works as expected
- t2 update: works as expected

To verify, do a `t2 push <file>` of some project that is known to work correctly with `t2 run <file>`

### CODE REVIEW

File: lib/tessel/deploy.js [hunk 1]
```diff
@@ -445,16 +445,21 @@ exportables.createShellScript = function(tessel, opts) {
       if (error) {
         return reject(error);
       }
-      // When the remote process finishes
-      remoteProcess.once('close', () => {
+
+      tessel.receive(remoteProcess, (error, received) => {
+        /* istanbul ignore if */
+        if (error && !error.message.includes('records in')) {
```
Comment (rwaldron) [inline lib/tessel/deploy.js]: Yes, I know this is awful.

**Comment by rwaldron** [comment]
image

**Comment by HipsterBrown** [comment]
Looks like there are some linting errors. Looks good otherwise. :+1:

**Comment by rwaldron** [comment]
> Does it set an exit code?

I thought to myself "ya know... I'm not sure, let's find out...". Turns out SSH2 does offer an exitCode if such exists, however it appears to be `undefined` in this case :|

**Comment by rwaldron** [comment]
I'm still working on revising this

**Comment by rwaldron** [comment]
@reconbot @HipsterBrown I'm using an exit code now. If there's an error, but an exit code = 0, then we'll ignore it.

**Pull request merged**